lib/types/PexCredentialMapper.ts

```diff
@@ -0,0 +1,303 @@
+
+  public static isSdJwtDecodedCredential(original: unknown): original is SdJwtDecodedVerifiableCredential {
+    return CredentialMapper.isSdJwtDecodedCredential(original as any);
+  }
+  public static isJwtDecodedCredential(original: unknown): original is JwtDecodedVerifiableCredential {
+    return CredentialMapper.isJwtDecodedCredential(original as any);
+  }
+
+  public static isSdJwtEncoded(original: unknown): original is string {
+    return CredentialMapper.isSdJwtEncoded(original as any);
+  }
+
+  public static isJwtEncoded(original: unknown): original is string {
+    return CredentialMapper.isJwtEncoded(original as any);
+  }
+
+  public static decodeVerifiableCredential(credential: OriginalVerifiableCredential, hasher?: HasherSync) {
+    return CredentialMapper.decodeVerifiableCredential(credential as any, hasher);
+  }
+
+  public static isCredential(original: OriginalVerifiableCredential | OriginalVerifiablePresentation): original is OriginalVerifiableCredential {
+    return CredentialMapper.isCredential(original as any);
+  }
+
+  public static areOriginalVerifiableCredentialsEqual(firstOriginal: OriginalVerifiableCredential, secondOriginal: OriginalVerifiableCredential) {
+    if (this.isMsoMdocDecodedCredential(firstOriginal) || this.isMsoMdocDecodedCredential(secondOriginal)) {
+      return uint8ArrayToBase64Url(cborEncode(firstOriginal)) === uint8ArrayToBase64Url(cborEncode(secondOriginal));
+    }
+
+    return CredentialMapper.areOriginalVerifiableCredentialsEqual(firstOriginal, secondOriginal);
+  }
+}
+
+/**
+ * Record where keys are the namespaces and the values are objects again with the namespace values
+ */
+export type MdocDecodedPayload = Record<string, Record<string, string | number | boolean>>;
+export function getMdocDecodedPayload(mdoc: IssuerSignedDocument): MdocDecodedPayload {
+  const namespaces = mdoc.issuerSigned.nameSpaces;
+
+  const decodedPayload: MdocDecodedPayload = {};
+  for (const [namespace, items] of Array.from(namespaces.entries())) {
+    decodedPayload[namespace] = items.reduce(
+      (acc, item) => ({
+        ...acc,
+        [item.elementIdentifier]: encodeMdocValue(item.elementValue),
+      }),
+      {},
+    );
+  }
+
+  return decodedPayload;
+}
+
+function encodeMdocValue(value: unknown) {
+  if (typeof value === 'string' || typeof value === 'number' || typeof value === 'undefined' || value === null) return value;
+  if (value instanceof Date || value instanceof DateOnly) return value.toISOString();
+
+  // TODO: we don't want undefined, but empty object might also not work?
+  return {};
+}
+
+function convertBase64urlToBinary(data: string) {
+  // Replace base64url characters back to base64
+  const base64 = data.replace(/-/g, '+').replace(/_/g, '/');
+
+  // Add padding if needed
+  const paddedBase64 = base64.padEnd(base64.length + ((4 - (base64.length % 4)) % 4), '=');
+
+  // Decode base64 to binary
+  const raw = atob(paddedBase64);
```

Comment:
Is atob supported in all environments?

Reply:
it seems to be supported in hermes starting from RN 0.74.